identityClient/identityClient.go

```diff
@@ -1211,3 +1211,28 @@ func (c *E3dbIdentityClient) AccessRequestGroups(ctx context.Context, params Acc
 	err = e3dbClients.MakeSignedServiceCall(ctx, c.requester, request, c.SigningKeys, c.ClientID, &groupsResponse)
 	return groupsResponse, err
 }
+
+// SyncFederatedIdentities gets information for all or a subset of federated Identities in a realm
+func (c *E3dbIdentityClient) SyncFederatedIdentities(ctx context.Context, params SyncFederatedIdentitiesRequest) (*SyncFederatedIdentitiesResponse, error) {
+	var federatedIdentities *SyncFederatedIdentitiesResponse
+	path := c.Host + identityServiceBasePath + "/" + federationResourceName + "/sync"
+	request, err := e3dbClients.CreateRequest("POST", path, params)
+	if err != nil {
+		return federatedIdentities, err
+	}
+	// Set Credential Token
+	for key, val := range params.Credentials {
+		var isAuthHeader bool
+		for _, authHeader := range AuthenticationHeaders {
```

Comment:
i'd find this more readable if you interchange these loops. looping over the AuthHeaders would eliminate the need for the 2nd loop altogether.

pseudocode:
```
for header in auth:
  val, exists := Credentials[header]
  if exists
    set request header to val
```